### ISSUE TITLE
Push-based notification stream

### ISSUE THREAD

**Issue by rraziel** [Bungie-net/api#1012]
I know this is not a small request at all, but I figured I might as well ask the question and see what folks at Bungie think of it: unless I missed that information in the documentation there is currently no push-based mechanism for the API, meaning the only way to get fresh information is to poll for it.

It would be quite nice to have some kind of streaming mechanism to receive events from the platform, via SSE, WebSocket or even MQTT.

This way an application could subscribe to the topic and get to know when:
- gear changes are made: swapping weapons in-game, etc.
- activities are started: going into a crucible match, a strike, a destination, etc.
- vendor inventory changes
- objectives/triumphs/bounties/quest step/etc. events
- new fireteam posted
- etc.

I am aware this kind of usage is probably more relevant for companion apps - which is what I am working on - than more general Destiny applications, but still think it would be very useful and possibly reduce the load on the platform when used properly to replace polling.

So basically my question is: what there ever any discussion about this, and if not, could we start one now?

Note: great API by the way, thanks a lot for offering it

**Comment by vthornheart-bng** [comment]
I would absolutely love that: it's something I've been dreaming about having for a long time.  Unfortunately, the infrastructure cost to create and maintain it has made it infeasible thus far.  I still hold out hope - and have an item in our backlog - about this idea, but I can't promise that we'll ever actually be able to make it happen with the time and resources (both for creation and maintenance/cost of operation) we're able to devote to the API.

However, if it ever does come to pass, I'll definitely be making a lot of noise about it here - so you're in the right place to keep an eye on it!

**Comment by yyewolf** [comment]
Bump, if it's still on tracks ?
